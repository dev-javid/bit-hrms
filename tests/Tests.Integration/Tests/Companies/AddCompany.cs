namespace Tests.Integration.Tests.Companies
{
    public class AddCompany(TestWebApplicationFactory<Program> factory) : IntegrationTest(factory)
    {
        private const string Route = "/api/companies";

        [Fact]
        [UseReporter(typeof(DiffReporter))]
        public async Task Post_Company()
        {
            await LoginAsSuperAdminAsync();

            var command = new
            {
                Name = Faker.Random.String(3, 100),
                AdministratorName = Faker.Random.String(3, 100),
                Faker.Person.Email,
                PhoneNumber = Faker.Random.Number(10000000, 999999999).ToString(),
            };

            HttpResponseMessage response;
            using (StringContent content = new StringContent(JsonSerializer.Serialize(command), Encoding.UTF8, JsoneMediaType))
            {
                response = await Client.PostAsync(Route, content);
            }

            response.StatusCode.Should().Be(HttpStatusCode.OK);
            var result = await response.Content.ReadAsStringAsync();
            Approvals.VerifyJson(result.Ignore("*"));
        }
    }
}
