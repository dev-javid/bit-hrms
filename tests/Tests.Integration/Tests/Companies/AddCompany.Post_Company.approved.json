{
    "weeklyOffDays": [
        "__IGNORED_VALUE__"
    ],
    "companyId": "__IGNORED_VALUE__",
    "name": "__IGNORED_VALUE__",
    "phoneNumber": "__IGNORED_VALUE__",
    "financialMonth": "__IGNORED_VALUE__",
    "administratorName": "__IGNORED_VALUE__",
    "email": "__IGNORED_VALUE__"
}